File: Homeworks/HW2/homework.m
function homework
f = @partial_sums
figure(1)
hold ON
fplot(f)
fplot(-1)
end


function [P] = partial_sums(t)
N = [1,2,4,8,16]
A = zeros(1,length(N))
    for i=1:length(N)
        A(i) = 2*(1-cos(N(i)*pi))/(N(i)*pi)*sin(2*N(i)*t)
    end
   P = sum(A) 
 %P = 2*(1-cos(N(1)*pi))/(N(1)*pi)*sin(2*N(1)*t)
end

File: Homeworks/HW2/homework.jl
t = 10;
n=2
sin(t)*sin(n*t)
1/2*(cos((n-1)*t)-cos((1+n)*t))
cos(3*pi)

sin(10)*cos(10)
sin(2*10)/2

A = zeros(5);
N = [1,2,4,8,16];
for n in 1:length(N)
    A[n] = (2*pi*N[n]*cos(pi*N[n])-2*sin(pi*N[n]))/(pi*N[n]*(pi*N[n]-sin(pi*N[n])))
end
using Plots

function FS(x)
    A[1] * sin(pi*N[1]*x) + A[2] * sin(pi*N[2]*x) +
    A[3] * sin(pi*N[3]*x)+ A[4] *  sin(pi*N[4]*x) +
    A[5] * sin(pi*N[5]*x)
 end

 f =plot(FS, title = "Fourier Series for sawtooth function", legend = false)


 savefig(f,"file.png")
